proxy/proxy/client.go

```diff
@@ -36,6 +37,10 @@ const (
 // user.
 var errNotCached = errors.New("instance was not found in cache")
 
+// connectionsCounter is used to enforce the optional maxConnections limit
+var connectionsCounter uint64 = 0
```

Comment:
This should be a field inside the Client struct instead of a package-level constant.